lib/dialog-renderer/index.tsx

```diff
@@ -1,64 +1,95 @@
+  themeClass: string;
+  isElectron: boolean;
+  isMacApp: boolean;
+};
+
+type StateProps = {
+  dialogs: T.DialogType[];
+};
+
+type DispatchProps = {
+  closeDialog: () => any;
+};
+
+type Props = OwnProps & StateProps & DispatchProps;
 
-    if (DialogComponent === null) {
-      throw new Error('Unknown dialog type.');
-    }
+export class DialogRenderer extends Component<Props> {
+  static displayName = 'DialogRenderer';
 
-    const closeThisDialog = () => closeDialog({ key });
+  render() {
+    const {
+      appProps,
+      buckets,
+      themeClass,
+      isElectron,
+      isMacApp,
+      closeDialog,
+    } = this.props;
 
     return (
-      <Modal
-        key={key}
-        className="dialog-renderer__content"
-        contentLabel={title}
-        isOpen
-        onRequestClose={closeThisDialog}
-        overlayClassName="dialog-renderer__overlay"
-        portalClassName={classNames('dialog-renderer__portal', themeClass)}
-      >
-        <DialogComponent
-          buckets={buckets}
-          dialog={dialog}
-          requestClose={closeThisDialog}
-          isElectron={isElectron}
-          isMacApp={isMacApp}
-          {...appProps}
-        />
-      </Modal>
+      <Fragment>
+        {this.props.dialogs.map(dialog => (
+          <Modal
+            key={dialog}
+            className="dialog-renderer__content"
+            contentLabel={dialog}
+            isOpen
+            onRequestClose={closeDialog}
+            overlayClassName="dialog-renderer__overlay"
+            portalClassName={classNames('dialog-renderer__portal', themeClass)}
+          >
+            {'ABOUT' === dialog ? (
+              <AboutDialog key="about" />
+            ) : 'IMPORT' === dialog ? (
+              <ImportDialog
+                key="import"
+                buckets={buckets}
+                isElectron={isElectron}
+              />
+            ) : 'SETTINGS' === dialog ? (
+              <SettingsDialog
+                key="settings"
+                buckets={buckets}
+                isElectron={isElectron}
+                isMacApp={isMacApp}
+                {...appProps}
+              />
+            ) : 'SHARE' === dialog ? (
+              <ShareDialog key="share" />
+            ) : null}
+          </Modal>
+        ))}
+      </Fragment>
     );
-  };
+  }
+}
 
-  return <Fragment>{dialogs.map(renderDialog)}</Fragment>;
-};
+const mapStateToProps: S.MapState<StateProps> = ({ ui: { dialogs } }) => ({
+  dialogs,
+});
 
-DialogRenderer.propTypes = {
-  appProps: PropTypes.object.isRequired,
-  buckets: PropTypes.object,
-  themeClass: PropTypes.string,
-  closeDialog: PropTypes.func.isRequired,
-  dialogs: PropTypes.array.isRequired,
-  isElectron: PropTypes.bool.isRequired,
-  isMacApp: PropTypes.bool.isRequired,
-};
+const mapDispatchToProps: S.MapDispatch<DispatchProps> = dispatch => ({
+  closeDialog: () => {
+    dispatch(closeDialog());
+  },
```

Comment:
just a nitpick, but we can collapse this one of two ways, or in two steps:

```ts
// the extra curlies aren't necessary
const mapDispatchToProps: S.MapDispatch<DispatchProps> = dispatch => ({
   closeDialog: () => dispatch(closeDialog())
});
```

```ts
// and we don't need to create a function here largely because our action-creators
// don't depend on additional parameters
const mapDispatchToProps: S.MapDispatch<DispatchProps> = { closeDialog };
```

at least the first step is nice. the second one is more for your choosing.